src/main/resources/manuals/funcs/__IS_ARRAY_INDEX__.ir

```diff
@@ -0,0 +1,25 @@
+def <AUX>:__IS_ARRAY_INDEX__(
+  P: Symbol | String
+): Boolean = {
+  if (= (typeof P) @String) {
+    call %0 = clo<CanonicalNumericIndexString>(P)
+    if (= %0 undefined) {
+      return false
+    } else {
+      // IsIntegralNumber AO will be removed in ES2025
+      call %1 = clo<IsIntegralNumber>(%0)
```

Comment:
here; see also: https://github.com/tc39/ecma262/pull/3344

Reply:
We don't have to use `IsIntegralNumber` AO because we already support the `is an array index` condition using `ETypeCheck` IR expression. So, I refactored the `__IS_ARRAY_INDEX__` aux function using it.